### PR TITLE
File summary return file id and file number

### PR DESCRIPTION
**Pull request by pbolduc**
# Description

This PR includes the following proposed change(s):

- Adds FileNumber to FileSummary. **Breaking Change**: FileId is the internal `Guid` value that should be used in API calls, the FileNumber is the human readable number that is displayed to the user.
- Adds Escaping of string values used in filter expressions to avoid injection attack
- Adds checking `id` or guid parameters are actually guid before attempting to query.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by KyleKayfish — APPROVED** [review]

**Review by KyleKayfish — CHANGES_REQUESTED** [review]
Update to resolve conflict

**Review by KyleKayfish — APPROVED** [review]

File: src/backend/Csrs.Api/Repositories/DynamicsExtensions.cs [hunk 1]
```diff
@@ -254,24 +265,5 @@ public static async Task<MicrosoftDynamicsCRMssgCsrsbccourtlevelCollection> GetC
             return values;
         }
 
-        public static async Task<string> GetSharepointDocumentLocationIdByRelatveUrl(this IDynamicsClient dynamicsClient, string relativeUrl, CancellationToken cancellationToken)
```
Comment (TayGov) [inline src/backend/Csrs.Api/Repositories/DynamicsExtensions.cs:257]: GetSharepointDocumentLocationIdByRelatveUrl is used and should not be deleted

**Review by TayGov — CHANGES_REQUESTED** [review]
Looks like a method got deleted

**Review by TayGov — APPROVED** [review]
👍

**Pull request merged**